tensorflow/python/ops/ragged/ragged_factory_ops.py

```diff
@@ -155,7 +156,7 @@ def _constant_value(ragged_factory, inner_factory, pylist, dtype, ragged_rank,
   if ragged_tensor.is_ragged(pylist):
     raise TypeError("pylist may not be a RaggedTensor or RaggedTensorValue.")
 
-  if not isinstance(pylist, (list, tuple)):
+  if np.ndim(pylist) == 0:
```

Comment:
It looks to me like np.ndim scans the entire input, which could be unnecessarily expensive.  How about changing this to:

    if np.isscalar(pylist) or not isinstance(pylist, (list, tuple, np.ndarray):

Reply:
The problem with this is that `np.isscalar(np.ndarray(1))` evaluates to `False`. How about short-circuiting `list` and `tuple`?
```
if not isinstance(pylist, (list, tuple)) and np.ndim(pylist) == 0:
```

Reply:
Sounds good to me. Thanks.